database/pgcsv_to_json

```diff
@@ -77,56 +77,12 @@ def convert_type_func(ty, ty_rest = ""):
     else:
       raise ValueError("Unsupported data type %s" % ty)
 
-# PostgreSQL COPY TO text Format parser
-# See: http://www.postgresql.org/docs/9.1/static/sql-copy.html#AEN64302
-pgTextEscapeSeqs = {
-    "b": "\b",
-    "f": "\f",
-    "n": "\n",
-    "r": "\r",
-    "t": "\t",
-    "v": "\v",
-    }
-def decode_pg_text_escapes(m):
-  c = m.group(1)
-  if c in pgTextEscapeSeqs:
-    return pgTextEscapeSeqs[c]
-  elif c.startswith("x"):
-    return chr(int(c, base=16))
-  elif c.startswith("0"):
-    return chr(int(c, base=8))
-  else:
-    return c
-def unescape_postgres_text_format(s):
-  # unescape PostgreSQL text format
-  return re.sub(r"\\(.|0[0-7]{1,2}|x[0-9A-Fa-f]{1,2})", decode_pg_text_escapes, s)
-
-def utf_8_encoder(unicode_csv_data):
-    for line in unicode_csv_data:
-        yield line.encode('utf-8')
-
-
-def unicode_csv_reader(unicode_csv_data, **kwargs):
-    '''A generator for unicode lines from a csv file.
-
-    This function takes all the same parameters as `csv.reader`. It was copied
-    directly from the documentation for that module:
-
-    https://docs.python.org/2/library/csv.html#examples
-
-    '''
-    # csv.py doesn't do Unicode; encode temporarily as UTF-8:
-    csv_reader = csv.reader(utf_8_encoder(unicode_csv_data), **kwargs)
-    for row in csv_reader:
-        # decode UTF-8 back to Unicode, cell by cell:
-        yield [unicode(cell, 'utf-8') for cell in row]
-
 def main():
   # parse column names and types from arguments
   def parseArg(arg):
       field_name, field_type = re.match(r"(.+):([^:]+)", arg).groups()
       return field_name, convert_type_func(field_type)
-  names_converters = map(parseArg, sys.argv[1:])
+  names_converters = list(map(parseArg, sys.argv[1:]))
 
   # read the PostgreSQL COPY TO text format
   # XXX With Python's csv.reader, it's impossible to distinguish empty strings from nulls in PostgreSQL's csv output.
```

Comment:
OK, I thought the `csv null as '\\N'` would address this ambiguity. I suppose this script is called from more than one place... Would the "FORCE_QUOTE" option of COPY also help here?

You have the best judgement of whether 1 or 2 is better.